### ISSUE TITLE
Issue while playing offline DRM videos

### ISSUE THREAD

**Issue by dushyantclassplus** [google/ExoPlayer#10815]
I'm trying to implement Offline DRM on my end, and the video is getting downloaded and working fine when I'm in the app. But when I close the app, close the internet, it's not able to play. 

My assumption is that the offline license is not getting downloaded. 

Adding gist for the code here:
Player: https://gist.github.com/dushyantclassplus/f97aa3bb95708172f6334e51af9d7d76
Download: https://gist.github.com/dushyantclassplus/21dea3d1ed8a95b0802c0b6d44e9a7ad

Please provide some info or help me guide in the right direction.

**Comment by icbaker** [comment]
It's a little hard to follow exactly how your example code is wired together - there seems to be some magic about how `content.assetId` (not sure what that name means?) is transmitted from `VideoDRMSessionMangaer` to `OfflineExoPlayer`.

I would check to make sure the `assetId` value you're decoding into the `MediaItem.drmConfiguration.keySetId` field is a value that you expect. Maybe put some code in there to see if it's a valid offline `keySetId` (e.g. maybe pass it directly to [`MediaDrm.getOfflineLicenseState`](https://developer.android.com/reference/android/media/MediaDrm#getOfflineLicenseState(byte[])) as a test).

> But when I close the app, close the internet, it's not able to play.

Without further information about the failure you observe there's not much more help we can offer I'm afraid.

**Comment by dushyantclassplus** [comment]
So I basically pass the assetId (or offlineAssetKeyId) to VideoDRMSessionMangaer from Download Tracker. Like this:

`drmSessionManager = VideoDrmSessionManager.createDrmSessionManager(
                                            contentId,
                                            jwSignatureData.getDrmUrls().getManifestUrl(),
                                            jwSignatureData.getDrmUrls().getLicenseUrl(), "");
                                    startDownloadDialogHelper = new StartDownloadDialogHelper(fragmentManager,
                                            getDownloadHelperForDRM(renderersFactory, jwSignatureData.getDrmUrls().getManifestUrl()),
                                            name, unsignedUri.toString());`
                                            
                                            
 Then save the data in DB and use it in the player.
 
 But the behaviour is weird, I'm able to play some of the drm videos offline and some are not working for some reason. Any particular reason why it could be happening?
 
 Working DRM Video Logs: https://gist.github.com/dushyantclassplus/95358e33971c7876f18d04e5d8a141ae
 
 Not Working DRM Video Logs: https://gist.github.com/dushyantclassplus/af59485ba9c6f6722b324d3b4177a722
